superscore/widgets/window.py

```diff
@@ -330,30 +344,78 @@ def open_pv_details(self, index: QtCore.QModelIndex, view: QtWidgets.QAbstractIt
             raise TypeError("Invalid model type passed to open_pv_details")
 
         # Get data via the client for alarm limits
-        epics_data: EpicsData
-        epics_data = self.client.cl.get(data.pv_name)
+        epics_data: EpicsData = None
+        try:
+            epics_data = self.client.cl.get(data.readback or data.setpoint)
+        except CAException as e:
+            logging.exception(e)
 
         pv_details = PVDetails(
-            pv_name=data.pv_name,
-            readback_name=data.readback.pv_name if getattr(data, "readback", None) else None,
+            pv_name=data.setpoint,
+            readback_name=data.readback,
             description=data.description,
-            tolerance_abs=data.abs_tolerance if isinstance(data, Parameter) else None,
-            tolerance_rel=data.rel_tolerance if isinstance(data, Parameter) else None,
+            tolerance_abs=data.abs_tolerance,
+            tolerance_rel=data.rel_tolerance,
             lolo=epics_data.lower_alarm_limit if isinstance(epics_data, EpicsData) else None,
             low=epics_data.lower_warning_limit if isinstance(epics_data, EpicsData) else None,
             high=epics_data.upper_warning_limit if isinstance(epics_data, EpicsData) else None,
             hihi=epics_data.upper_alarm_limit if isinstance(epics_data, EpicsData) else None,
             tags=data.tags,
         )
-        self.popup = PVDetailsPopup(tag_groups=self.client.backend.get_tags(), pv_details=pv_details)
+        popup_class = PVDetailsPopupEditable if editable else PVDetailsPopup
+        self.popup = popup_class(
+            tag_groups=self.client.backend.get_tags(),
+            pv_details=pv_details,
+            pv_id=data.uuid
+        )
+        if editable:
+            self.popup.accepted.connect(self.update_pv)
+            self.popup.accepted.connect(lambda: view.model().sourceModel().refetch_row(index.row()))
         self.popup.adjustSize()
 
         table_top_right = view.mapToGlobal(view.rect().topRight())
-
         x = table_top_right.x() - self.popup.width()
         y = table_top_right.y()
-
         self.popup.move(x, y)
+
+        self.popup.show()
+
+    @QtCore.Slot()
+    def update_pv(self):
+        pv_id = self.sender().pv_id
+        pv_details = self.sender().pv_details
+        self.client.backend.update_pv(
+            pv_id,
+            pv_name=pv_details.pv_name,
+            description=pv_details.description,
+            tags=pv_details.tags,
+            abs_tolerance=pv_details.tolerance_abs,
+            rel_tolerance=pv_details.tolerance_rel,
+        )
+
+    @QtCore.Slot(QtWidgets.QAbstractItemView)
```

Comment:
```suggestion
    @QtCore.Slot()
```

This slot decorator expected different parameters from what the function expected. It threw an error for me, but this change fixed it. I'm surprised the slot signature compatibility would be this strict.

Reply:
Could you do me a favor and try adding a PV using the `squirrel-alpha` env? The code works as-is for my personal env and for `squirrel-alpha`, so I'm wondering if it's a `PyQT` / `PySide` version issue.

Reply:
@shilorigins The error did NOT come up when using the `squirrel-alpha` environment.

Reply:
Thanks for checking. Since the method does in fact not accept a parameter, I've made the requested change, but we should keep in mind that the slot behaviour can be unpredictable and maybe take a look at why your env behaves differently.